internal/future/app/mmlogic/mmlogic_service.go

```diff
@@ -15,31 +15,139 @@
 package mmlogic
 
 import (
+	"time"
+
+	"encoding/json"
+
+	"github.com/gomodule/redigo/redis"
+	"github.com/sirupsen/logrus"
+	"open-match.dev/open-match/internal/config"
 	"open-match.dev/open-match/internal/future/pb"
+	"open-match.dev/open-match/internal/future/storage"
+	"open-match.dev/open-match/internal/set"
+
+	"fmt"
+)
+
+var (
+	logger = logrus.WithFields(logrus.Fields{
+		"app":       "openmatch",
+		"component": "app.mmlogic",
+	})
 )
 
 // The MMLogic API provides utility functions for common MMF functionality such
 // as retreiving Tickets from state storage.
 type mmlogicService struct {
+	storageService storage.Service
+	cfg            config.View
 }
 
 // RetrievePool gets the list of Tickets that match every Filter in the
 // specified Pool.
 // TODO: Consider renaming to "GetPool" to be consistent with HTTP REST CRUD
 // conventions. Right now there's a GET and a POST for this verb.
-func (s *mmlogicService) RetrievePool(req *pb.RetrievePoolRequest, stream pb.MmLogic_RetrievePoolServer) error {
-	ctx := stream.Context()
+func (s *mmlogicService) RetrievePool(req *pb.RetrievePoolRequest, responseServer pb.MmLogic_RetrievePoolServer) error {
+	poolName := req.Pool.Name
+	poolFilters := req.Pool.Filter
+	redisConn := storage.GetDeprecatedRedisPool(s.storageService).Get()
+	defer func() {
+		if err := redisConn.Close(); err != nil {
+			logger.WithError(err).Error("Failed to close Redis connection")
+		}
+	}()
+
+	logger.WithFields(logrus.Fields{
+		"Pool.Name":    poolName,
+		"Pool.Filters": poolFilters,
+	}).Debug("Received RetrievePoolRequest.")
+
+	// A map[attribute]map[playerID]value
+	attributeToTickets := make(map[string]map[string]int64)
+	// A set of playerIds that satisfies all filters
+	idSet := make([]string, 0)
+
+	// For each filter, do a range query to Redis on Filter.Attribute
+	for i, filter := range poolFilters {
+		startTime := time.Now()
+		// Time Complexity O(logN + M), where N is the number of elements in the attribute set
+		// and M is the number of entries being returned.
+		idToValue, err := redis.Int64Map(redisConn.Do("ZRANGEBYSCORE", filter.Attribute, filter.Min, filter.Max, "WITHSCORES"))
+
+		if err != nil {
+			logger.WithFields(logrus.Fields{
+				"Command": fmt.Sprintf("ZRANGEBYSCORE %s %f %f WITHSCORES", filter.Attribute, filter.Min, filter.Max),
+			}).WithError(err)
```

Comment:
If this query fails why are we continuing?
Also we should respond with a reasonable gRPC error code.